### PR TITLE
MINIFICPP-1074 - TailFileTests was not working on WIN32 due do CLRF

### PR DESCRIPTION
**Pull request by asdaraujo**
Several points in the test cases implementation were not handling
the WIN32 CRLF enf-of-line correctly.

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/standard-processors/tests/unit/TailFileTests.cpp [hunk 1]
```diff
@@ -41,11 +41,44 @@
...
-        "four,five,six, seven";
-static const char *TMP_FILE = "minifi-tmpfile.txt";
-static const char *STATE_FILE = "minifi-state-file.txt";
+#ifdef WIN32
+int CR_LEN = 1;
+#else
+int CR_LEN = 0;
+#endif
+
+static std::string FIRST_LINE = "one,two,three";  // NOLINT
```
Comment (arpadboda) [inline extensions/standard-processors/tests/unit/TailFileTests.cpp]: Can you make these const, too? That would also resolve the linter issues here.

Comment (asdaraujo) [reply extensions/standard-processors/tests/unit/TailFileTests.cpp]: Even after making it const I still get linter errors:
```
For a static/global string constant, use a C style string instead: "static const char FIRST_LINE[]"
```

Is it ok to leave it as a `const std::string` and with the `// NOLINT` directive?

Comment (arpadboda) [reply extensions/standard-processors/tests/unit/TailFileTests.cpp]: Yes, I'm ok with that.

**Comment by asdaraujo** [comment]
@arpadboda Anything else needed from me here? The error in Travis seems to be unrelated to the changes (`No CMAKE_CXX_COMPILER could be found.`)

**Comment by arpadboda** [comment]
Closing this a #731 is going to resolve this testcase issue as well.

**Pull request closed without merging**